### PR TITLE
Make distance of gps-new-entity-place available as soon as it's added

### PR DESCRIPTION
**Pull request by nickw1**
**⚠️ All PRs have to be done versus 'dev' branch, so be aware of that, or we'll close your issue ⚠️**

**What kind of change does this PR introduce?**

Allows the `distance` property of `gps-new-entity-place`s to be available as soon as they are added.
**Can it be referenced to an Issue? If so what is the issue # ?**

**How can we test it?**

Try out the `show-distance` example from A-Frame `new-location-based` examples. Previously, you had to set the location initially (to create the boxes) and then set it again to access the `distance` property. Now the `distance` property is available as soon as the boxes are added (you need to input a location for them to appear, then click on them to test).

**Summary**

This solves issue #502. It works by reintroducing the `gps-entity-place-added` event, which was available in the older location-based components. The `gps-new-camera` receives this event, and sends its own location back to the `gps-new-entity-place` to set its distance.

**Does this PR introduce a breaking change?**

No

**Please TEST your PR before proposing it. Specify here what device you have used for tests, version of OS and version of Browser**

Firefox latest, Linux, desktop. 

**Other information**

### CODE REVIEW

**Comment by kalwalt** [comment]
@nickw1 tested your code, i don't know if an issue when testing in developer mode i got this error in the chrome console:
```
Uncaught ReferenceError: msg is not defined
    at i._displayError (aframe-ar.js:1:1638714)
    at Object.gpserror (aframe-ar.js:1:1636796)
    at _watchPositionId._watchPositionId.navigator.geolocation.watchPosition.enableHighAccuracy (aframe-ar.js:1:783100)
```

**Comment by nickw1** [comment]
@kalwalt thanks for finding that. I've now fixed it - this is an old bug, not related to this commit!

File: aframe/src/new-location-based/gps-new-camera.js [hunk 1]
```diff
@@ -143,9 +143,9 @@ AFRAME.registerComponent("gps-new-camera", {
   _displayError: function (error) {
     const arjs = this.el.sceneEl.systems["arjs"];
     if (arjs) {
-      arjs._displayErrorPopup(msg);
```
Comment (kalwalt) [inline aframe/src/new-location-based/gps-new-camera.js:134]: this make sense!

Comment (kalwalt) [reply aframe/src/new-location-based/gps-new-camera.js:134]: this fix error reported in [comment](https://github.com/AR-js-org/AR.js/pull/507#issuecomment-1430150668)

Comment (kalwalt) [reply aframe/src/new-location-based/gps-new-camera.js:134]: @nickw1 this is fixed i tested again and the error is not shown anymore.

**Review by kalwalt — APPROVED** [review]
@nickw1 It's fine for me, we can merge it if you want.

File: aframe/src/new-location-based/gps-new-camera.js (hunk 1 above)
Comment (nickw1) [reply aframe/src/new-location-based/gps-new-camera.js:134]: @kalwalt ok great, thanks.

Comment (kalwalt) [reply aframe/src/new-location-based/gps-new-camera.js:134]: with Typescript the compiler will alert you, and as a consequence this kind of bugs are solved immediately.

**Pull request merged**